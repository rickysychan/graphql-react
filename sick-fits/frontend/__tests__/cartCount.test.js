import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CartCount from "../components/CartCount";

describe("<cartCount />", () => {
  it("should render correctly", () => {
    shallow(<CartCount count={10} />);
  });

  it("should match the snapshot", () => {
    const cartCount = shallow(<CartCount count={10} />);
    expect(toJSON(cartCount)).toMatchSnapshot();
  });

  it("should ", () => {
    const cartCount = shallow(<CartCount count={50} />);
    expect(toJSON(cartCount)).toMatchSnapshot();
    cartCount.setProps({ count: 10 });
    expect(toJSON(cartCount)).toMatchSnapshot();
  });
});
