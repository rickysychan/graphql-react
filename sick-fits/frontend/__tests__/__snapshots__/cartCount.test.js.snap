// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<cartCount /> should  1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        50
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<cartCount /> should  2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<cartCount /> should match the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
